wdl/gwas/saige_sub.multiproc.wdl

```diff
@@ -0,0 +1,141 @@
+task test {
+
+	File nullfile
+	File varianceratiofile = sub(nullfile, ".rda", ".varianceRatio.txt")
+    String outfileprefix = basename(nullfile, ".rda") + "-"
+	Array[File] bgenfiles
+	File samplefile
+	Int minmac
+	String docker
+    String loco
+
+    command {
+
+        python3 <<EOF
+        import os
+        import shlex
+        import subprocess
+        processes = set()
+        cmd_prefix = 'step2_SPAtests.R \
+            --minMAC=${minmac} \
+            --sampleFile=${samplefile} \
+            --GMMATmodelFile=${nullfile} \
+            --varianceRatioFile=${varianceratiofile} \
+            --numLinesOutput=1000 \
+            --IsOutputAFinCaseCtrl=TRUE \
+            --LOCO=${loco} '
+        for file in '${sep=" " bgenfiles}'.split(' '):
+            cmd = cmd_prefix + '--bgenFile=' + file
+            cmd = cmd + ' --SAIGEOutputFile=${outfileprefix}' + os.path.basename(file) + '.SAIGE.txt'
+            logfile = open('SAIGE_log_${outfileprefix}' + os.path.basename(file) + '.txt', 'w')
+            processes.add(subprocess.Popen(shlex.split(cmd), stdout=logfile))
+        for p in processes:
+            if p.poll() is None:
```

Comment:
is both poll and wait really needed or could you just wait?

Reply:
Can just wait

Reply:
actually... maybe would be better to poll for all jobs and then sleep for a minute and poll again. This way any failed job can be identified and the whole task could throw an error. Now we may be waiting for the first task to finish and some processes might silently fail...